src/fs.c

```diff
@@ -233,7 +233,7 @@ int fs_file_write(struct fs_file *f, char *buffer, uint32_t n)
 	const struct fs_dirent_ops *ops = d->ops;
 	uint32_t i, total_copy_length = 0;
 	uint32_t direct_addresses_start = bm->block, direct_addresses_end =  bm->block + (bm->offset + n) / bm->block_size;
-	uint32_t start_offset = bm->offset % bm->block_size, end_offset = (bm->offset + n - 1) % bm->block_size;
```

Comment:
Are these bugs in kevinfs independent of the feature in this PR?
If so, make a separate pull request for those bug fixes.